src/back/spv/writer.rs

```diff
@@ -1145,20 +1230,24 @@ impl Writer {
                             crate::TypeInner::Vector { size, kind, width } => {
                                 vector_id = Some(left_id);
                                 for (k, v) in self.lookup_type.iter() {
```

Comment:
why are we not just issuing a lookup into the map? we shouldn't need to iterate the arenas anywhere more than once, really

Reply:
I completely agree, this is probably the most dirtiest piece of code in this back-end. I will take a look to fix it, so it won't iterate the arena again. This piece of code will be completely refactored in the future I presume, as it only supports very basic cases.

Reply:
Oh and by the way, as I see now, LocalType's arent even supported here. I will fix that too.